lib/Map/TableStructure.js

```diff
@@ -451,8 +451,10 @@ function calculateFinishDatesFromStartDates(startJulianDates, localDefaultFinalD
         var n = sortedUniqueJulianDates.length;
         if (n > 1) {
             finalDurationSeconds = JulianDate.secondsDifference(sortedUniqueJulianDates[n - 1], sortedUniqueJulianDates[0]) / (n - 1);
+            endDates.push(JulianDate.addSeconds(sortedUniqueJulianDates[n - 1], finalDurationSeconds, new JulianDate()));
+        } else {
+            endDates.push(undefined);
```

Comment:
@kring Is this the right thing to do for `n === 1`?

Reply:
😬 I think so. The way `endDates` is used below, it looks like there has to be something added, or the indices will be wrong.